src/lustre/lustreInput.ml

```diff
@@ -332,21 +342,35 @@ let of_channel old_frontend only_parse in_ch =
                 s |> HString.string_of_hstring |> LustreIdent.mk_string_ident)
           )
         in
+        let main_nodes = match Flags.lus_main_type () with
+          | Some s -> 
+            let s_ident = LustreIdent.mk_string_ident s in 
+            if not (List.mem `CONTRACTCK (Flags.enabled ())) then
+              let msg =
+                Format.asprintf "Option --lus_main_type can only be used when realizability checking is enabled (--enable CONTRACTCK)"
+              in
+              raise (MainTypeWithoutRealizability msg)
+            else (
+              try 
+                let s_ident = LustreIdent.mk_string_ident (LGI.type_tag ^ s) in
+                let _ = LN.node_of_name s_ident nodes in  
+                match Flags.lus_main () with 
+                | Some _ -> s_ident :: main_nodes
+                | None -> [s_ident]
+              (* User-specified type alias in command-line input might not exist *)
+              with Not_found -> 
+                let msg =
+                  Format.asprintf "No refinement type with alias '%a' found in input"
+                    (LustreIdent.pp_print_ident false) s_ident
+                in
+                raise (NoMainNode msg)
+            )
+          | None -> main_nodes in
         Ok (nodes, globals, main_nodes)
     in
 
     match result with
     | Ok (nodes, globals, main_nodes) ->
-      (* Check that main nodes all exist *)
-      main_nodes |> List.iter (fun mn ->
-        if not (LN.exists_node_of_name mn nodes) then
-          (* This can only happen when the name is passed as command-line argument *)
-          let msg =
-            Format.asprintf "Main node '%a' not found in input"
-              (LustreIdent.pp_print_ident false) mn
-          in
-          raise (NoMainNode msg)
-      ) ;
```

Comment:
This also covered the case where `old_frontend` is true. This is no longer the case with the new location.

Reply:
Addressed by handling --lus_main and --lus_main_type analogously in the old frontend. 

Instead of fully repeating the checks performed in the new frontend, I did not support --lus_main_type in the old frontend, nor did I fully support checking Kind 2-generated imported nodes in the old frontend. I did this simply for brevity (which we may value more than supporting all the features, since we don't think the old front-end is being used nowadays); if you want, I can easily update the code to fully support these features.

Reply:
Sounds good. Thanks.